### ISSUE TITLE
Graph stays flat although the sensor values have changed over time

### ISSUE THREAD

**Issue by ivanovd** [kalkih/mini-graph-card#933]
After the 2023.0 update of HA the graphs on the cards started showing a flat line, even though all sensors were updating regularly.

Here is the card config:
<img width="760" alt="image" src="https://user-images.githubusercontent.com/4889583/222880066-d5de8783-244d-4d3d-9fa5-bbd9c9a4277c.png">

Here is the sensor history for the past 24 hours:
<img width="417" alt="image" src="https://user-images.githubusercontent.com/4889583/222880088-9c241ce1-5ffd-43e7-a5ee-a36155f617b3.png">

As you can see, this happens for all sensors ... I have another instance where I have only two mini cards with the same configuration and they work fine:
<img width="779" alt="image" src="https://user-images.githubusercontent.com/4889583/222880137-e22cf301-3f01-488d-bdf3-9760610fc76e.png">
<img width="411" alt="image" src="https://user-images.githubusercontent.com/4889583/222880143-452e7bf2-62ea-4733-8540-46541133790b.png">

So I am not sure if it's a bug related to the number of stacked cards or it is something else ... did clear the cache and hard reset few times but that didn't help.

Any ideas are welcome.

**Comment by raphv** [comment]
I have the exact same issue. I have upgraded from 2023.2.4 to 2023.3.1 yesterday evening and every graph now looks like a plateau! (See below with a history card for comparison)
![IMG_20230304_083746](https://user-images.githubusercontent.com/1217834/222883051-598437bb-8da6-4a26-a26f-08318687a653.jpg)

**Comment by ildar170975** [comment]
@ivanovd 
Your code does not have points_per_hour specified and no info about actual sensor’s update frequency is provided.

**Comment by ildar170975** [comment]
@raphv 
No code is posted, nothing to analyze ((

**Comment by raphv** [comment]
Hi @ildar170975 let me give more details.
First, restarting again *does* solve the issue, however historical data between the update and the restart is still incorrect.
Secondly this issue only affects state values, attributes were correctly.
Finally, changing the points per hours (in the case below from 1 to 2) fixes the incorrect historical data issue. Please find below 2 screenshots: at 1 point per hour there is a green flat line:
![IMG_20230304_135530](https://user-images.githubusercontent.com/1217834/222903055-9b86c47e-6872-456f-966f-135e4fbdda50.jpg)
At 2 points per hour, the green line does change.
![IMG_20230304_135338](https://user-images.githubusercontent.com/1217834/222903190-381e4cf3-8880-4310-baa3-242221b620f8.jpg)
I'm seeing this issue on multiple sensors
```
type: custom:mini-graph-card
name: Outdoors Air Quality
hours_to_show: 24
points_per_hour: 2
aggregate_func: last
hour24: true
line_width: 2
show:
  fill: fade
  icon: false
entities:
  - entity: sensor.waqi_location
    name: Overall
    unit: US AQI
  - entity: sensor.waqi_location
    name: NO₂
    attribute: nitrogen_dioxide
    color: '#0000cc'
    show_fill: false
  - entity: sensor.waqi_location
    name: O₃
    attribute: ozone
    color: '#00cccc'
    show_fill: false
  - entity: sensor.waqi_location
    name: PM10
    attribute: pm_10
    color: '#a0a0a0'
    show_fill: false
  - entity: sensor.waqi_location
    name: PM25
    attribute: pm_2_5
    color: '#606060'
    show_fill: false
color_thresholds:
  - value: 0
    color: '#00ff00'
  - value: 75
    color: '#cccc00'
  - value: 150
    color: '#ff0000'
  - value: 300
    color: '#ff00ff'

**Comment by ivanovd** [comment]
> @ivanovd Your code does not have points_per_hour specified and no info about actual sensor’s update frequency is provided.

While that is true. It was working before 2023.1 and it works on my other instance as you can see. I will try with the suggested options. The sensors are ZigBee therefore they send update on valu change, not on regular intervals.

**Comment by ildar170975** [comment]
@ivanovd 
The fact that it worked does not prove that your setup is correct. 
Number of points per hour should be at least same as a maximum frequency or

**Comment by raphv** [comment]
@ivanovd @ildar170975 I've had the issue with sensors that are updated every 30 seconds (temperature, pressure), every 3 minutes (local air quality sensors) and every 5 minutes (the WAQI API, although typically the value itself is only updated hourly), so the frequency doesn't seem to me to be the problem. The custom graph card just flatlined in all cases, while the standard history graph didn't.

**Comment by ildar170975** [comment]
@raphv 
What your provided is a card with several sensors , one of them is flatlined. And no conventional history-graph card for the same data.

**Comment by raphv** [comment]
@ildar170975
Can you be clearer on what information exactly you miss. I have created a history graph to show you. Also, I believe this is some form of cache issue, but it's not the browser cache because I have the issue from multiple browsers. Also, what is interesting is that if I duplicate the card and then make **any** change, the data shows correctly. In the screenshot below, only the first line has incorrect flat segments.
![Screenshot_2023-03-04-16-24-31-172_io homeassistant companion android](https://user-images.githubusercontent.com/1217834/222914687-abe07ce3-711c-4709-9956-9ccfc6cf85d4.jpg)

**Comment by ildar170975** [comment]
Assume that you see the XXXX sensor flatlined. 
Then please show a mini graph card with THIS sensor only. 
And history graph for THIS sensor. 
Same hours-to-show. 
Pontss-per-hour - equal to or x2 of real update frequency. 

Sorry for typos, posting from iPhone.

**Comment by raphv** [comment]
You can see exactly that information in my post above: the green line is correct in the second case and incorrect in the first, and the third one shows it at the highest possible frequencies. Since there is this issue on multiple sensors (see my posts above) with very different update frequencies, I don't really understand what else I can provide to help you diagnose the issue.

**Comment by ildar170975** [comment]
Asked about it several times. 
Make one more post with this info: https://github.com/kalkih/mini-graph-card/issues/933#issuecomment-1454785577
If you have a problem with one sensor - do not confuse people with unrelated information. Show info for this particular sensor only.

**Comment by raphv** [comment]
But the problem, and I've tried to explain it multiple times, is that if I remove other sensors and change the configuration of the card, then the problem disappears and I have nothing wrong to show you!

**Comment by muth-zhaw** [comment]
Unfortunately I have the same issue and I will try to give you the requested information: 

The sensor state sensor.heizung_waermezaehler_mk1_stand is updated every 5 minutes (12 times per hour).
This is the card configuration:
```
type: custom:mini-graph-card
name: Heizung Zählerstand MK1
entities:
  - entity: sensor.heizung_waermezaehler_mk1_stand
    name: MK1
    show_state: true
    points: false
hours_to_show: 120
points_per_hour: 6
cache: true
hour24: true
```
The graph looks like this:
![image](https://user-images.githubusercontent.com/25752001/222917993-a78267cf-29dd-4a39-9811-ef8c66d60634.png)

As you can see the state is displayed correctly, but the graph has is flat since yesterday (Fri, 03-03-2023 11:35). The upper boundary shows the value (106,794) but the actual state is 106,880.

To show you that the values are not missing, here is the "show more" graph:
![image](https://user-images.githubusercontent.com/25752001/222918305-f642c148-68b0-44d9-90ef-0522a859a7ea.png)

This are the last values from the database:
![image](https://user-images.githubusercontent.com/25752001/222919183-91ee00f8-c414-4872-8af0-2b6caad2de06.png)

I would be very grateful for a fix.

**Comment by ildar170975** [comment]
@raphv 
Probably if we discuss this issue via Discord or Telegram, we would solve it faster. Otherwise we deal with language problem, misunderstanding etc. 

First - two things which I would like to explain. If you do understand this - very good. 
But anyway just in case:

1.Assume you have a sensor which has a state and an attribute. 
Using history graph you may build a graph for the state only. 
To build a graph for the attribute - a template sensor must be created. 
Using mini graph you may build a graph for the attribute too - but in case of any problems with the graph (like a flatline) it is hard to investigate since there is nothing to compare with (i.e. history graph for this attribute). That is why I never use mini graph for attributes, always create template sensors for them. 

2.There is a difference between history graph and mini graph. History graph shows exact changes of a state. Mini graph shows a calculated curve which is defined based on options like points per hour, smoothing etc. To make a mini graph show a curve which is close to a real curve - some options should be set properly, like smoothing = false, points per hour = at least same as a real update frequency (or 2x, 3x, …). For my “Speedtest” graphs (measured once an hour) I anyway use points per hour = 60 - just to have a better presentation of my graph. 

As for your posts - what I noticed:

1.Regarding this post: https://github.com/kalkih/mini-graph-card/issues/933#issuecomment-1454646494
Here I can see 2 mini graphs and 2 history graphs. 
Since there is no code - hard to say what could be wrong. 

2.Starting from this post: https://github.com/kalkih/mini-graph-card/issues/933#issuecomment-1454731821
What I understand is one graph is flatlined. Which one particularly - was unclear, you have several lines. You said “green flatline” - but there are 2 green-ish lines))). That is why I asked “show me a card with ONE bad sensor” - to be show which is a problematic one. Then I asked you to provide a history graph for this data. Since I do not know what is a name of a problematic sensor (I do not need a long friendly name, I need exact unique entity_id) and no code was provided for the history graph - again I may only guess.
Whenever someone needs to report a problem - he should provide a detailed info which may be used by another person to reproduce the problem.

**Comment by raphv** [comment]
Hello again,
I am sorry that I cannot provide all the details. Restarting Home Assistant and changing details of the mini-graph cards was enough to fix my issue, but it also means that I have lost access to the data you need. I was hoping that the descriptions I was providing next to it would help as well, but reading your reply, it seems that I wasn't clear enough.
Again, I cannot show you a card with ONE bad sensor, because I've lost access to these (or it is otherwise not very obvious since the outage was at a time where some sensors had limited variation). I showed you a complex and hard to understand case indeed, but you have to believe me that I've encountered this issue on several sensors that are very different from each others, in very different configurations of the mini-graph cards: some used the `average` aggregating function, others the `last`, some had 2 points per hour, others 0.5.
*Every single sensor* was showing as a flat line between the time I upgraded and the time I restarted, and the code for each card was very different.
For example, the code below also gave a flat line, but it wouldn't help if I show you what it looks like since it no longer is an error:
```
type: custom:mini-graph-card
name: Temperature
hour24: true
decimals: 1
hours_to_show: 48
points_per_hour: 0.5
show:
  extrema: true
  fill: fade
entities:
  - sensor.pws_temperature
```
Changing anything to the code, even a colour, removes the flat line, while reverting to the exact original configuration shows the flat line again.
With regards to the attributes, you can completely ignore them since (again) they are working perfectly fine and it was only the 'overall' line (which is a state sensor) that was flat. But again, I had to include them in the screenshot and code, since making any changes to the configuration hides the problem.
Since my personal configuration is now working, I suggest that you focus on other people who've commented above, I was just trying to provide additional details, but apparently they weren't the right ones.

**Comment by raphv** [comment]
Another example, if this helps you.
The two cards in the screenshot have almost exactly the same configuration, the only difference is a colour.
The card at the bottom shows the correct atmospheric pressure, the one at the top has a flat line that shouldn't be here.
![IMG_20230304_213621](https://user-images.githubusercontent.com/1217834/222927602-671fe092-c5e9-4256-b91a-25c6014d2963.jpg)
The code for the card at the top (incorrect):
```
type: custom:mini-graph-card
name: Atmospheric Pressure
hours_to_show: 48
hour24: true
points_per_hour: 0.5
decimals: 0
show:
  extrema: true
  fill: fade
entities:
  - sensor.nettigo_air_monitor_bme280_pressure
color_thresholds:
  - value: 900
    color: '#0000ff'
  - value: 983
    color: '#0000cc'
  - value: 1013
    color: '#cc00cc'
  - value: 1043
    color: '#0000cc'
  - value: 1100
    color: '#0000ff'
```
The card at the bottom (correct):
```
type: custom:mini-graph-card
name: Atmospheric Pressure
hours_to_show: 48
hour24: true
points_per_hour: 0.5
decimals: 0
show:
  extrema: true
  fill: fade
entities:
  - sensor.nettigo_air_monitor_bme280_pressure
color_thresholds:
  - value: 900
    color: '#0000ff'
  - value: 983
    color: '#0000cc'
  - value: 1013
    color: '#cc00cc'
  - value: 1043
    color: '#0000cc'
  - value: 1100
    color: '#0000fc'
```
Do you have an idea what may be causing such a difference?

**Comment by ildar170975** [comment]
First, I am not a developer of this card, but have some experience with it and trying to help people here with it. The author of this card is not maintaining the project anymore. There are a few Dev persons who sometimes suggest changes, a few users like me who help to investigate issues. 
As for the last example - it seems to be like a caching problem…. Could you clean a browser cache to be sure?
Also, as I said before, I usually suggest to always post a history graph to exclude the Recorder’s issues…

**Comment by MattJeanes** [comment]
I'm also seeing this issue since upgrading HA to 2023.3.1:

![image](https://user-images.githubusercontent.com/2363642/223434126-163f2e2c-f510-4d95-aa8d-3418feed6ed5.png)

![image](https://user-images.githubusercontent.com/2363642/223434182-37ef727d-349d-468e-9972-51855e7ea0df.png)

Very strange!

**Comment by ildar170975** [comment]
@MattJeanes 
No additional info is provided. Read my posts above, correct your code if needed.

**Comment by MattJeanes** [comment]
Apologies @ildar170975 should have provided more info

Here is the configuration of the card:

```yaml
type: custom:mini-graph-card
entities:
  - entity: sensor.growatt_battery_state_of_charge
    name: Battery state of charge (past 24 hours)
show:
  points: hover
  labels: true
hours_to_show: 24
points_per_hour: 6
lower_bound: 10
upper_bound: 100
color_thresholds:
  - value: 40
    color: '#43a047'
  - value: 30
    color: '#f39c12'
  - value: 20
    color: '#c0392b'
```

I have tried changing it to a minimal version but this is still showing the same issue:

```yaml
type: custom:mini-graph-card
entities:
  - entity: sensor.growatt_battery_state_of_charge
    name: Battery state of charge (past 24 hours)
```

![image](https://user-images.githubusercontent.com/2363642/223456683-109ce30c-2623-49ba-8e03-5693caf6416d.png)

Here is the history view as well showing that sensor

![image](https://user-images.githubusercontent.com/2363642/223456968-223ffd38-a6f2-4253-979a-3f026b75089e.png)

I have cleared my browser cache and tried other browsers and devices but they all show the same issue. Interestingly the current state does show correctly as you can see it shows 11% but the graph is not showing anywhere near that for some reason.

If there is any more information I can provide please let me know, the only relevant log in dev tools console is `MINI-GRAPH-CARD  0.11.0`.

I could try restarting HA but if it works then I won't be able to help diagnose the issue until it re-appears again.

**Comment by ildar170975** [comment]
> Here is the history 

No, edit your post and give a CARD (not a History view) with same `hours_to_show`.



> ```yaml
> lower_bound: 10
> upper_bound: 100
> color_thresholds:
> ```

Remove for testing.

**Comment by MattJeanes** [comment]
@ildar170975 I have updated my comment above with the new information.

**Comment by ildar170975** [comment]
Very strange.
Usually - and it happens several times a month - I see flatlines too, but `history-graph`s are flatlined too, this is a Recorder's issue. The only choice I have then is a reboot.
When only `mini-graph`s are flatlined, I could only suspect a cache problem, but you tested on several devices...
Well, no ideas, unfortunately.
Reboot and keep living...

**Comment by MattJeanes** [comment]
Rebooting HA 'fixed' it as I suspected 

![image](https://user-images.githubusercontent.com/2363642/223486201-e28e3478-62bb-4c21-93a8-8ad7ccb15ea7.png)

Hopefully this doesn't happen too often, I am curious where the bug is - HA or here, but in any case that's sorted it for the moment at least. Will be keeping an eye on the bug fixes list for the next HA patch 😄

**Comment by bobloadmire** [comment]
i've rebooted the host system multiple times, still not displaying correctly since updating.

**Comment by MattJeanes** [comment]
> 

That's the ability within HA to adjust the displayed precision of sensors introduced in 2023.3 right? No I haven't used this yet

**Comment by bobloadmire** [comment]
> 

I thought of that, but changing precision didn't make a difference

**Comment by ildar170975** [comment]
What HA version are you on?

**Comment by harristom** [comment]
Same issue here on 2023.03.1

Card:
`type: custom:mini-graph-card
name: Valves
entities:
  - entity: sensor.thermostat_living_room_1_valve_state
    name: Living Room 1
  - entity: sensor.thermostat_living_room_2_valve_state
    name: Living Room 2
  - entity: sensor.thermostat_living_room_3_valve_state
    name: Living Room 3
  - entity: sensor.thermostat_bathroom_valve_state
    name: Bathroom
  - entity: sensor.thermostat_kitchen_valve_state
    name: Kitchen
  - entity: binary_sensor.study_valve_state
    name: Study
    y_axis: secondary
  - entity: binary_sensor.bedroom_valve_state
    name: Bedroom
    y_axis: secondary
state_map:
  - value: 'off'
    label: Closed
  - value: 'on'
    label: Open
points_per_hour: 12
aggregate_func: max
line_width: 2.5
lower_bound: 0
upper_bound: 100
lower_bound_secondary: 0
upper_bound_secondary: 1
`

You can see that Living Room 3 shows 9% at 11:35-39 but actually was at 0% the whole time. Living Room 3 is just an example, the other valves on the chart have the same problem and indeed other mini graph cards are also exhibiting the issue.

![2023-03-09 11_45_11-Home Assistant - Home – Home Assistant](https://user-images.githubusercontent.com/51212431/224001703-328d6d31-fd20-4181-b3c8-30065b5eea90.png)

![2023-03-09 11_45_52-Home Assistant - Home – Home Assistant](https://user-images.githubusercontent.com/51212431/224001724-c3cf0cb5-3483-4112-a4a4-0f1e3bd136a1.png)

**Comment by diamant-x** [comment]
Same issue here.
Appeared after upgrading to any of the new 2023.3 versions. (at least right after installing x.2 and x.3).
Data seems to be recorded correctly, is just the custom chart that doesn't load them.
Clearing cache didn't work, other devices didn't work.
Only workaround is a second restart of HA.
Logs don't show anything. Not sure if there's anyway to obtain traces of the custom card like with integrations?
![image](https://user-images.githubusercontent.com/7346338/224263192-fca9a598-3aba-40aa-8dc5-816b1217cf58.png)

**Comment by antsalm** [comment]
This flatlining has been present since 2023.3. Atleast for me. I run HA in a container. Sometimes rebooting Home Assistant is enough, sometimes rebooting whole container fixes it. It affects all devices, so CTRL+F5 does not fix it. I think it gets stuck on initial values of the used sensor.

Picture show how it briefly worked one of the reboots this morning as I was working with other stuff on HA.

![image](https://user-images.githubusercontent.com/20434971/224263782-4b060ed6-a233-4c6b-afb3-472ce2d57438.png)


```
type: custom:mini-graph-card
name: L1
aling_header: left
animate: true
hours_to_show: 2
points_per_hour: 30
hour24: true
line_width: 6
font_size: 78
align_state: center
decimals: 0
update_interval: 2
show:
  extrema: false
  icon: false
upper_bound_secondary: 2500
lower_bound_secondary: 0
entities:
  - entity: sensor.l1_power
    show_fill: true
    color: '#9e6b1b'
    name: L1
```

**Comment by Wummeke** [comment]
Same issue here.

![image](https://user-images.githubusercontent.com/22944993/224424449-7d689dd2-c9a8-4180-b9ab-e0543c59e469.png)

Both mini-graph-card and history card show the last 24 Hours.
here are the configs:

```
- animate: true
  color_thresholds:
    - color: '#a55eea'
      value: -6.9
    - color: '#8854d0'
      value: -3.9
    - color: '#3867d6'
      value: -1.2
    - color: '#4b7bec'
      value: 1.6
    - color: '#2d98da'
      value: 4.3
    - color: '#45aaf2'
      value: 7.1
    - color: '#0fb9b1'
      value: 9.9
    - color: '#2bcbba'
      value: 12.7
    - color: '#20bf6b'
      value: 15.5
    - color: '#26de81'
      value: 18.2
    - color: '#fed330'
      value: 21
    - color: '#f7b731'
      value: 23.8
    - color: '#fd9644'
      value: 26.6
    - color: '#fa8231'
      value: 29.3
    - color: '#fc5c65'
      value: 32.1
    - color: '#eb3b5a'
      value: 35.9
  entities:
    - entity: sensor.outside_temperature
      index: 0
      name: 'Balkon'
      show_state: true
      state_adaptive_color: true
    - color: '#9f9f9f'
      entity: sensor.solar_elevation
      line_width: 1
      show_fill: false
      show_legend: false
      show_state: false
      y_axis: secondary
  hour24: true
  align_icon: left
  line_width: 2
  lower_bound_secondary: 0
  name: 'Balkon'
  points_per_hour: 12
  hours_to_show: 24
  show:
    graph: line
    points: false
  type: 'custom:mini-graph-card'

- type: history-graph
    title: "Balkon temperature"
    hours_to_show: 24
    entities:
      - entity: sensor.outside_temperature
        name: "Balkon"
```

After a restart of HA it works for a few hours, but then this happens again.

**Comment by Ascathon** [comment]
For me it works fine on my Android phone with the app...
But on the desktop I had to restart hass to get it working. Changing settings like points her hour to a different value "fixes" it but changing it back to desired value gives me the flatliner again.
So probably some cache but exactly.

**Comment by Ascathon** [comment]
"Fixed" it by removing a node in the local storage.
![2023-03-11-00_13_51](https://user-images.githubusercontent.com/21990236/224445545-420efac5-0733-4248-800c-46b912c2950f.png)

**Comment by canedje** [comment]
here also problems. No history shown in the graphs.
running HA 2023.3.3 supervised

**Comment by Wummeke** [comment]
I just tried different browsers, A fresh never been used before Firefox shows the same behaviour as in Chrome and on mobile. A fresh never been used before Edge also shows flatlines. 

FireFox:
![image](https://user-images.githubusercontent.com/22944993/224490721-e5a5c604-4140-4282-8bf1-81e003ea90c3.png)

Edge:
![image](https://user-images.githubusercontent.com/22944993/224490877-1a4c8039-c26c-400f-b38c-0ebc68825fed.png)

Chrome: 
![image](https://user-images.githubusercontent.com/22944993/224490900-b1d98cd4-d65a-4240-8c47-7c0eb05beb38.png)

(config of the Balkon sensor is posted above, the Licht sensor is similar)

**Comment by ildar170975** [comment]
Once again. Already wrote about it earlier but still see same posts.

1. One issue = 1 post. No "wrote above/earlier", all info in ONE place.
2. One post must contain this info:
-- minimal `mini-graph-card` config to demonstrate an issue, w/o color_thresholds & state_map, for only ONE sensor which is flatlined:
```
type: custom:mini-graph-card
entities:
  - entity: sensor.my_sensor
show:
  labels: true
hours_to_show: 24 ## not too big & not too small - to show that 2/3 of time the graph is OK, then became flatlined
points_per_hour: 60 ### at least equal to the real update interval or x2
smoothing: false
aggregate_func: last
line_width: 1
lower_bound: 0 ### or any other value to see a flatline
```
-- a screenshot of that `min-graph-card` (default light theme);
-- a conventional `history graph` for the same `hours_to_show` for the same ONE sensor - not a History page, not a more-info (read above why more-info may be WRONG)
-- a screenshot of that `history-graph` (default light theme);
-- info about sensor - how often is it polled (a real update freq);
-- when you started observing flatlines, did it stop after HA reboot or after clearing a browser's cache (also, test with 'cache: false' option).

**Comment by MattJeanes** [comment]
It seems clear to me that this is a bug introduced by an update of HA to 2023.3 (or maybe one of the patch versions), but the question remains of if the issue goes away completely after a HA reboot or if it returns later.

Personally, I have not seen the issue reoccur in the past couple days since rebooting, and I have applied a further update to 2023.3.3 without any issues too but I'm still waiting to see if it does come back.

If a reboot is all that's needed to fix it and it stays fixed then that's probably good enough and no further fix is required, maybe a little note in the readme or pinning this issue wouldn't go amiss but not super important.

If however it's going to keep coming back then it's going to need a fix somewhere, whether that's in HA itself or here I'm not sure but I think we should wait and see if a reboot fixes it permanently first.

It could be a bug in HAs API that mini-graph-card is using or it could be an implementation detail of something that used to work accidentally but now doesn't that mini-graph-card relied on. 🤷

So in summary, let's reboot and let live until we see it come back, then look into it further if it does.

**Comment by Der-Sven** [comment]
I can confirm, that the issues occur after updating to 23.3. DId that yesterday night and since the the graph-card draws wrongly.

What I have seen, is that for all my graph cards the same (not to the last digit after the comma) wrong values are beeing used for drawing.  no matter what type of sensor.

<img width="123" alt="Screenshot 2023-03-12 121320" src="https://user-images.githubusercontent.com/76218102/224540989-46e95e96-4071-4f2c-b76f-6767c15f6440.png">
<img width="120" alt="Screenshot 2023-03-12 121340" src="https://user-images.githubusercontent.com/76218102/224540991-3bf50a57-0e2a-4de0-8897-5befd049aa78.png">
<img width="243" alt="Screenshot 2023-03-12 121401" src="https://user-images.githubusercontent.com/76218102/224540992-4d2e1332-db47-4289-9146-67fc923a87bc.png">

Rebooting solved it for now. I will monitor ans see if it reuccurs after time.

**Comment by Crayjin** [comment]
Same here. After reboot flatline started again. Here Reboot was done on 12.03.2023 at 22:08ish

![image](https://user-images.githubusercontent.com/24563614/224689252-0dfc0e0c-f35a-4aa4-aac4-e09b9711949e.png)
![image](https://user-images.githubusercontent.com/24563614/224689322-bba351d0-c7a5-4b74-a0bf-2fd196e383b6.png)

**Comment by adrianrdzv** [comment]
Yeah the issue is definitely related to the points per hour value.
My card is displaying 12 hours, here are the screenshots with `points_per_hour`

**1**
<img width="169" alt="Screenshot 2023-03-13 at 10 24 20" src="https://user-images.githubusercontent.com/4851625/224764200-bdbbc086-2416-4848-ade9-897006dfc4f8.png">

**2**
<img width="169" alt="Screenshot 2023-03-13 at 10 24 28" src="https://user-images.githubusercontent.com/4851625/224764239-b6af2724-5a4a-4ead-9019-0c80ddf5c8ef.png">

**3**
<img width="167" alt="Screenshot 2023-03-13 at 10 24 38" src="https://user-images.githubusercontent.com/4851625/224764277-c10aec0d-d615-432d-be84-4bc3e7e9128f.png">

So originally I had 2, changing it to any other value fixed it.

**Comment by sperglord8008s** [comment]
Same Issue.

**Comment by ildar170975** [comment]
I clearly explained [here ](https://github.com/kalkih/mini-graph-card/issues/933#issuecomment-1464932117)what kind of posts are expected here.
Still see same posts /o needed info.
Probably some cases are related to "I do not know how to use the card" issue, but not possible to find out it - due to absence of info & ignoring requirements by users.
Due to this - all posts after [that post](https://github.com/kalkih/mini-graph-card/issues/933#issuecomment-1464932117) will be HIDDEN - unless users EDIT them accordingly to requirements.
Edit your post, write here smth like "post edited" - then the post will be displayed.
Any questions - ask here.

**Comment by raphv** [comment]
But can we have more posts about solutions? We've got a dozen users mentioning similar issues, but all we hear so far is "you are not providing the exact information that is needed" and no action being taken to address what seems to be a fairly common and important issue.

**Comment by ildar170975** [comment]
Solutions or recommendations may only be provided after analysis.

**Comment by raphv** [comment]
With at least 10 different people giving different levels of information about the same issue, what are you still lacking?

**Comment by ildar170975** [comment]
An answer is obvious. We are not using crystal bowls here.
Stop this, please.

**Comment by raphv** [comment]
No, it's not obvious. You've kept sending us running around in circles, and so far, out of the whole conversation, you have some cases documented in great detail and still no solution.

**Comment by ildar170975** [comment]
> You've kept sending us running around in circles

Is it not may problem that YOU do not know how to use it.
That is why YOU running in cycles.
Do not say for other people.

**Comment by ildar170975** [comment]
@raphv
Please tell me are you still experiencing the issue after HA reboot?
Not saying that "reboot solves the issue", just for analysis.

**Comment by SgtBatten** [comment]
@ildar170975 I can finally access github again. Thanks for your assistance going over this issue with me on the community forums. https://community.home-assistant.io/t/lovelace-mini-graph-card/71385/2294

I did not realise others were affected, so I agree this is likely an issue with HA and not the graph card. 

For anyone following along, my experience was that I could rectify the graph by changing any single parameter. eve editing the name of the graph caused the issue to vanish. 
I could also reproduce the error only on existing charts. If i copy the exact code to a new dashboard it would occur, but any changes would fix it. 

As it stands I only have one graph which goes back more than a day so it still shows the flatline still, all others have rectified  since updating to 2023.3.3 as their hours showing has passed. Very odd one.

**Comment by Thlerry** [comment]
Looks like I am having the exact same issue, but on in the HA app on iOS.
This is my graph in chrome on Mac:
![image](https://github.com/kalkih/mini-graph-card/assets/99030879/83cb0e69-b2da-4dc6-b176-e8e7f44c1892)

This exact same graph showing in the HA app with some flat lines:
![image](https://github.com/kalkih/mini-graph-card/assets/99030879/c076e350-ccc4-4fca-bd72-16cd63bd21a0)

It was working fine in the APP and suddenly the lines became flat at some spots.
Like others have mentioned editing the yaml and making a change fixes the issue for a bit, then it reappears.

`type: custom:mini-graph-card
name: AC Voltage AN (Line 1)
icon: mdi:home
show:
  extrema: false
  fill: false
  state: false
  name: false
  icon: false
entities:
  - entity: sensor.solaredge_ac_voltage_an
    name: Voltage AN
    color: '#ffaaaa'
  - entity: sensor.solaredge_ac_voltage_bn
    name: Voltage BN
    color: '#ff0088'
  - entity: sensor.solaredge_ac_voltage_cn
    name: Voltage CN
    color: '#00ffff'
  - entity: sensor.tb_volt_253
    name: Limit 253V
    color: rgba(255,255,255,0.2)
height: 250
line_width: 1
hour24: true
hours_to_show: 3
points_per_hour: 60
`

**Comment by ildar170975** [comment]
Check v0.14.0-dev.2
